### DISCUSSION TITLE
ESP-01S Stateless Button not working

### DISCUSSION THREAD

**Discussion by shengweitw** [RavenSystem/esp-homekit-devices#2100] [MEPLHAA Scripts]
In the old version, I used the following JSON, which worked.
{"c":{"o":0},"a":[{"t":3,"b":[],"s":0,"f0":[{"g":0,"t":1,"p":1}],"f1":[{"g":0,"t":2,"p":1}],"f2":[{"g":0,"t":3,"p":1}]},{"t":3,"b":[],"s":0,"f0":[{"g":2,"t":1,"p":1}],"f1":[{"g":2,"t":2,"p":1}],"f2":[{"g":2,"t":3,"p":1}]}]}
I use the following code, the first button works fine, but the second button doesn't respond?
Can someone help me?
From the beginning, I use the following JSON
{"c":{"io":[[[0],6],[[2],6]],"b":[[0,5],[2,5]]},"a":[{"t":3,"f0":[[0,1]],"f1":[[0,2]],"f2":[[0,3]]},{"t":3,"f0":[[2,1]],"f1":[[2,2]],"f2":[[2,3]]}]}
Then I tried using the following
{"c":{"io":[[[0,2],6]],"b":[[0,5],[2,5]]},"a":[{"t":3,"f0":[[0,1]],"f1":[[0,2]],"f2":[[0,3]]},{"t":3,"f0":[[2,1]],"f1":[[2,2]],"f2":[[2,3]]}]}
{"c":{"io":[[[0,2],6]],"b":[[[0,2],5]]},"a":[{"t":3,"f0":[[0,1]],"f1":[[0,2]],"f2":[[0,3]]},{"t":3,"f0":[[2,1]],"f1":[[2,2]],"f2":[[2,3]]}]}

But it doesn't work. I compared it with Wiki example for Stateless Button, Can someone tell me where I'm going wrong?
{"c":{"io":[[[0],6]],"b":[[0,5]]},"a":[{"t":3,"f0":[[0,1]],"f1":[[0,2]],"f2":[[0,3]]}]}

**Comment by Tellicious** [comment]
Try this:

`{"c":{"io":[[[14],3],[[0,2],6,1,1]],"l":14,"b":[[0,5],[2,5]]},"a":[{"t":3,"f0":[[0]],"f1":[[0,2]],"f2":[[0,3]]},{"t":3,"f0":[[2]],"f1":[[2,2]],"f2":[[2,3]]}]}`

**Reply by shengweitw** [reply]
use sample on GPIO0 is working

{"c":{"io":[[[0],6]],"b":[[0,5]]},"a":[{"t":3,"f0":[[0,1]],"f1":[[0,2]],"f2":[[0,3]]}]}

but if add GPIO2 is not work, i don't know why. Any suggestions?

{"c":{"io":[[[0],6],[[2],6]],"b":[[0,5],[2,5]]},"a":[{"t":3,"f0":[[0,1]],"f1":[[0,2]],"f2":[[0,3]]},{"t":3,"f0":[[2,1]],"f1":[[2,2]],"f2":[[2,3]]}]}

**Reply by Tellicious** [reply]
Are you using external pull-ups/pull-downs? if not, you have to use internal ones as I already suggested in my previous answer. Most likely GPIO0 already has its own external PU onboard, while GPIO2 doesn’t

**Reply by shengweitw** [reply]
Thank you very much. I forgot that GPIO2 needs a pull-up resistor. I used the following settings and it worked.

{"c":{"io":[[[0],6],[[2],6,1]],"b":[[[0],5]]},"a":[{"t":3,"f0":[[0,1]],"f1":[[0,2]],"f2":[[0,3]]},{"t":3,"f0":[[2,1]],"f1":[[2,2]],"f2":[[2,3]]}]}
